tests/director/tests/notifications/fixtures_test.go

```diff
@@ -342,72 +344,96 @@ func resetShouldFailEndpointFromExternalSvcMock(t *testing.T, client *http.Clien
-	samlAssertionDestBytes := client.GetDestinationByName(t, serviceURL, samlAssertionDestinationName, instanceID, token, http.StatusOK)
-	var samlAssertionDest esmdestinationcreator.SAMLAssertionDestination
+func assertSAMLAssertionDestination(t *testing.T, client *clients.DestinationClient, serviceURL, samlAssertionDestinationName, samlAssertionCertName, samlAssertionDestinationURL, app1BaseURL, instanceID, ownerSubaccountID, authToken, userTokenHeader string, expectedCertNames []string) {
+	samlAssertionDestBytes := client.FindDestinationByName(t, serviceURL, samlAssertionDestinationName, authToken, userTokenHeader, http.StatusOK)
+	var samlAssertionDest esmdestinationcreator.DestinationSvcSAMLAssertionDestResponse
 	err := json.Unmarshal(samlAssertionDestBytes, &samlAssertionDest)
 	require.NoError(t, err)
-	require.Equal(t, samlAssertionDestinationName, samlAssertionDest.Name)
-	require.Equal(t, directordestinationcreator.TypeHTTP, samlAssertionDest.Type)
-	require.Equal(t, samlAssertionDestinationURL, samlAssertionDest.URL)
-	require.Equal(t, directordestinationcreator.AuthTypeSAMLAssertion, samlAssertionDest.Authentication)
-	require.Equal(t, directordestinationcreator.ProxyTypeInternet, samlAssertionDest.ProxyType)
-	require.Equal(t, app1BaseURL, samlAssertionDest.Audience)
-	require.Equal(t, samlAssertionCertName+directordestinationcreator.JavaKeyStoreFileExtension, samlAssertionDest.KeyStoreLocation)
-}
-
-func assertClientCertAuthDestination(t *testing.T, client *clients.DestinationClient, serviceURL, clientCertAuthDestinationName, clientCertAuthCertName, clientCertAuthDestinationURL, instanceID, token string) {
-	clientCertAuthDestBytes := client.GetDestinationByName(t, serviceURL, clientCertAuthDestinationName, instanceID, token, http.StatusOK)
-	var clientCertAuthDest esmdestinationcreator.ClientCertificateAuthenticationDestination
+	require.Equal(t, ownerSubaccountID, samlAssertionDest.Owner.SubaccountID)
+	require.Equal(t, instanceID, samlAssertionDest.Owner.InstanceID)
+	require.Equal(t, samlAssertionDestinationName, samlAssertionDest.DestinationConfiguration.Name)
+	require.Equal(t, directordestinationcreator.TypeHTTP, samlAssertionDest.DestinationConfiguration.Type)
+	require.Equal(t, samlAssertionDestinationURL, samlAssertionDest.DestinationConfiguration.URL)
+	require.Equal(t, directordestinationcreator.AuthTypeSAMLAssertion, samlAssertionDest.DestinationConfiguration.Authentication)
+	require.Equal(t, directordestinationcreator.ProxyTypeInternet, samlAssertionDest.DestinationConfiguration.ProxyType)
+	require.Equal(t, app1BaseURL, samlAssertionDest.DestinationConfiguration.Audience)
+	require.Equal(t, samlAssertionCertName+directordestinationcreator.JavaKeyStoreFileExtension, samlAssertionDest.DestinationConfiguration.KeyStoreLocation)
+
+	require.Equal(t, len(expectedCertNames), len(samlAssertionDest.CertificateDetails))
+	for i := 0; i < len(expectedCertNames); i++ {
```

Comment:
How do we know if the certificarte details will always have that order

Reply:
Currently, I am not aware of any use-case where 1 Destination can have more than 1 Certificate. So, it's probably safe to assume that the length of that array will be 1.

However, in the future there could be a change which will require support of multiple certs, so I can re-write it with maps to be on the safe side. What do you think?

Reply:
If there wont be more than one certificate we can leave it as is. If usecase with more than one cert comes we will adapt the test and can adapt the fixture as well.